OpenRA.Mods.Common/Activities/Resupply.cs

```diff
@@ -78,18 +78,29 @@ public override bool Tick(Actor self)
 				if (aircraft != null)
 				{
 					aircraft.AllowYieldingReservation();
-					if (aircraft.Info.FlightDynamics.HasFlag(FlightDynamic.TakeOffOnResupply))
-						Queue(new TakeOff(self));
 
-					return true;
+					// ReturnToBase only handles TakeOffOnResupply when Resupply finishes, not when it's cancelled prematurely
+					if (aircraft.Info.FlightDynamics.HasFlag(FlightDynamic.TakeOffOnResupply))
```

Comment:
The `TakeOff` should be suppressed when `Resupply` is called by `ReturnToBase` with `alwaysLand = true`. This can be done by adding an `alwaysStay` parameter to the constructor and having `ReturnToBase` pass `alwaysLand` to that.